rust/rsc/src/bin/rsc/read_job.rs

```diff
@@ -203,3 +206,58 @@ pub async fn read_job(
         }
     }
 }
+
+#[tracing::instrument(skip_all)]
+pub async fn allow_job(
+    Json(payload): Json<AllowJobPayload>,
+    conn: Arc<DatabaseConnection>,
+    target_load: f64,
+    system_load: Arc<RwLock<f64>>,
+) -> StatusCode {
+    let hash = payload.hash();
+
+    match job::Entity::find()
+        .filter(job::Column::Hash.eq(hash.clone()))
+        .one(conn.as_ref())
+        .await
+    {
+        // Unable to run the query to lookup the job
+        Err(err) => {
+            tracing::error!(%err, "Failed to search for cached job");
+            return StatusCode::INTERNAL_SERVER_ERROR;
+        }
+        // Job is cached, don't try again
+        Ok(Some(_)) => return StatusCode::CONFLICT,
+        // Job is not cached, use the other deciding factors
+        Ok(None) => {}
+    }
+
+    // Reject a subset of jobs that are never worth caching
+    if payload.runtime < 0.1 {
+        return StatusCode::NOT_ACCEPTABLE;
+    }
+
+    // Statistically reject jobs when the system should shed load
+    let current_load = match system_load.read() {
+        Ok(lock) => *lock,
+        Err(err) => {
+            tracing::error!(%err, "Unable to lock system load for reading. Returning target load");
+            target_load
```

Comment:
I wonder if there is some edge case here where the machine becomes to loaded to read this and then gets stuck accepting all requests. Perhaps in the failure to read we should assume current_load requires some shedding?

Reply:
how do you feel about making a 50% load shed chance in that case?

Reply:
That seems reasonable to me.